include/nuttx/arch.h

```diff
@@ -1939,28 +1928,6 @@ int16_t up_fetchsub16(FAR volatile int16_t *addr, int16_t value);
 int8_t up_fetchsub8(FAR volatile int8_t *addr, int8_t value);
 #endif
 
-/****************************************************************************
- * Name: up_cpu_index
- *
- * Description:
- *   Return an index in the range of 0 through (CONFIG_SMP_NCPUS-1) that
- *   corresponds to the currently executing CPU.
- *
- * Input Parameters:
- *   None
- *
- * Returned Value:
- *   An integer index in the range of 0 through (CONFIG_SMP_NCPUS-1) that
- *   corresponds to the currently executing CPU.
- *
- ****************************************************************************/
-
-#ifdef CONFIG_SMP
```

Comment:
let's keep the prototype here, you can implement up_cpu_index as inline or not inline function freely with the prototype

Reply:
The implementation of up_interrupt_context() is different on different arch platform, especially some architectures that do not support SMP yet